src/nnet3/nnet-chain-example.cc

```diff
@@ -421,5 +421,49 @@ void ShiftChainExampleTimes(int32 frame_shift,
   }
 }
 
+
+void SelectFeatureOffset(int32 feature_offset, NnetChainExample *chain_eg) {
+  std::vector<NnetIo>::iterator iter = chain_eg->inputs.begin(),
+    end = chain_eg->inputs.end();
+  int32 ivec_dim, num_offsets;
+  Vector<BaseFloat> offset_vec;
+  for (; iter != end; ++iter) {
+    if (iter->name == "offset") {
+      num_offsets = iter->features.NumRows();
+      feature_offset = feature_offset % num_offsets;
+      Matrix<BaseFloat> offsets;
+      iter->features.GetMatrix(&offsets);
+      offset_vec.Resize(offsets.NumCols());
+      offset_vec.CopyRowFromMat(offsets, feature_offset);
+      break;
+    }
```

Comment:
Should there be an error if "offset" is not found?
Or at least a warning and a return statement.